Lib/test/test_builtin.py

```diff
@@ -2063,8 +2061,9 @@ def test_type_name(self):
                 self.assertEqual(A.__module__, __name__)
         with self.assertRaises(ValueError):
             type('A\x00B', (), {})
-        with self.assertRaises(ValueError):
-            type('A\udcdcB', (), {})
+        # TODO: RUSTPYTHON (https://github.com/RustPython/RustPython/issues/935)
+        # with self.assertRaises(ValueError):
+        #    type('A\udcdcB', (), {})
```

Comment:
We always prefer `expectedFailure` decorator than commenting out. (Guess what happens when this is fixed later)

Reply:
Yeah, that's a very fair point. In this case, I've fixed all but these failures, which are due to a much broader bug than I can reasonably fix alongside all these other failures. With a decorator, none of these assertions would run, so regressions could occur in all of the ones that I've just fixed. Should I move these assertions to a separate test to decorate? Or perhaps I can use it as a context manager? What do you think?

Reply:
OK, I've just pushed up a fix: use `self.assertRaises(AssertionError)` as a context manager around the expected failures: once they're passing, that will fail the test.

Reply:
We just left it or added a test in `extra_tests`, but this is an interesting approach.